include/nil/blueprint/parser.hpp

```diff
@@ -746,37 +746,41 @@ namespace nil {
                 std::uint32_t start_row = assignments[currProverIdx].allocated_rows();
 
                 // extract zk related metadata
-                const std::string metadataStr = extract_metadata(inst);
-                std::uint32_t userProverIdx = currProverIdx;
-                try {
-                    userProverIdx = std::stoi(metadataStr);
-                } catch(...) {
-                    userProverIdx = currProverIdx;
-                }
+                if (!frame.next_prover) {
+                    const std::string metadataStr = extract_metadata(inst);
+                    std::uint32_t userProverIdx = currProverIdx;
+                    try {
```

Comment:
Could we avoid using exceptions here? They are too expensive to check them on each instuction.

Reply:
done. By the way, argument of `#pragma` should be checked on compiler step (make sure if it integer), here just double check.